chef/automate
# Firewall API

### :nut_and_bolt: Description: What code changed, and why?
As a user I need to check the port reachability between two nodes on given port and using a protocol from Bastion, so that I do not face any issues while doing Automate HA deployment regarding firewall setup and my set up is as per the Automate HA requirements.



### :chains: Related Resources
https://chefio.atlassian.net/browse/CHEF-2296

### :+1: Definition of Done

### :athletic_shoe: How to Build and Test the Change
- Install the following version of automate-cli or create a new automate-cli with this branch and install it
- `ssanju/automate-cli/0.1.0/20230605104431`
- start go-fiber server with `chef-automate verify server`
- From postman make the following request to the endpoint `http://localhost:7799/api/v1/checks/firewall`
- **POST Request**:
```
{
  "source_node_ip": "",
  "destination_node_ip": "",
  "destination_service_port": "",
  "destination_service_protocol": "",
  "root_cert": ""
}
```
### :white_check_mark: Checklist

**All PRs** must tick these:


With occasional exceptions, all PRs **from Progress employees** must tick these:


All PRs **from Progress employees** should tick these if appropriate:


*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
<img width="1585" alt="Screenshot 2023-06-05 at 4 16 42 PM" src="https://github.com/chef/automate/assets/101255220/618afcaa-871e-4d4d-9acc-c309292f211e">
<img width="1706" alt="Screenshot 2023-06-05 at 4 17 16 PM" src="https://github.com/chef/automate/assets/101255220/4d0d6b4a-6a90-4309-ad5b-75a6a7e7e26d">
<img width="1702" alt="Screenshot 2023-06-05 at 4 17 38 PM" src="https://github.com/chef/automate/assets/101255220/d28927e9-e14e-4e0b-9ddb-6a55b88409b5">
<img width="1626" alt="Screenshot 2023-06-05 at 4 06 00 PM" src="https://github.com/chef/automate/assets/101255220/24551616-2b52-4383-9550-33bfa71c2c4c">
<img width="1634" alt="Screenshot 2023-06-05 at 4 06 24 PM" src="https://github.com/chef/automate/assets/101255220/57558ea3-6656-461c-9d3d-1a6a5ee69902">
<img width="1636" alt="Screenshot 2023-06-05 at 4 06 43 PM" src="https://github.com/chef/automate/assets/101255220/24743a41-164c-4eb9-aa81-6d94e90f1853">
<img width="1629" alt="Screenshot 2023-06-05 at 4 07 07 PM" src="https://github.com/chef/automate/assets/101255220/2fb509c3-0d67-4f52-87c5-108f45ee6a9c">
<img width="1629" alt="Screenshot 2023-06-05 at 4 07 41 PM" src="https://github.com/chef/automate/assets/101255220/246038ee-588d-4121-86c8-7c04ff79f154">
<img width="1627" alt="Screenshot 2023-06-05 at 4 07 59 PM" src="https://github.com/chef/automate/assets/101255220/cc8939f8-2f8c-49d3-bc7e-b1eb92a5bc0f">
<img width="1635" alt="Screenshot 2023-06-05 at 4 08 39 PM" src="https://github.com/chef/automate/assets/101255220/cf0626d1-ff44-41d1-83bb-cc300586b19c">
<img width="1641" alt="Screenshot 2023-06-05 at 4 09 43 PM" src="https://github.com/chef/automate/assets/101255220/19e92c30-8bc8-4a08-ae5e-5d31a3d3889e">
<img width="1628" alt="Screenshot 2023-06-05 at 4 10 10 PM" src="https://github.com/chef/automate/assets/101255220/9690da3a-5833-4561-aa41-39a3c9935ea2">
